code/modules/food/customizables.dm

```diff
@@ -659,6 +659,60 @@
 /obj/item/weapon/reagent_containers/food/snacks/customizable/candy/coin
 	name = "flavored chocolate coin"
 	icon_state = "coincustom"
+	var/sideup = COIN_HEADS //heads, tails or on its side?
+
+/obj/item/weapon/reagent_containers/food/snacks/customizable/candy/coin/New()
+	..()
+	pixel_x = rand(-8, 8) * PIXEL_MULTIPLIER
+	pixel_y = rand(-8, 0) * PIXEL_MULTIPLIER
+	sideup = pick(COIN_HEADS, COIN_TAILS)
+
+/obj/item/weapon/reagent_containers/food/snacks/customizable/candy/coin/proc/coinflip(var/mob/user, thrown, rigged = FALSE)
+	var/matrix/flipit = matrix()
```

Comment:
This seems to be copy-pasted from `/obj/item/weapon/reagent_containers/food/snacks/chococoin/proc/coinflip`. Can you avoid it?

Reply:
Should I define it for `/obj/item/weapon/reagent_containers/food/snacks/`? If so, that'd mean adding a `sideup` var to all snacks. What's preferable? Unless there is a way to check whether or not a var exists.
All of them come from `/obj/item/weapon/coin/`, but I think defining these procs and vars things for `/obj/item/weapon/` might be too general? Or would that be okay?

Reply:
How about a `/datum/component/coin`? It would define `coinflip()`.
The regular coins' `attack_self()`, provided everything super-calls, invokes `/event/item_attack_self`. So the coin component listens for that and calls `coinflip()`. 
For both the chocolate coin and the regular coins, the component would listen for an `/event/thrown` and call `coinflip()` with the right parameters.

Note that while currently no such thing as `/event/thrown` exists, it's simple to implement. Look at how https://github.com/vgstation-coders/vgstation13/blob/77d84446d5f47e1e7fa97a733912f83786c202f2/code/_onclick/item_attack.dm#L6 does it. Define the event, document it, invoke it at the beginning of the base proc, then have the component listen for it and react, like uplinks do here:
https://github.com/vgstation-coders/vgstation13/blob/77d84446d5f47e1e7fa97a733912f83786c202f2/code/datums/components/uplink.dm#L19

Reply:
You can ask me if you get stuck, but I'm quicker to respond on Discord.

Reply:
I think I got it to work, thanks!